### PR TITLE
Revert "clean and fix string split"

### PR DESCRIPTION
**Pull request by howardjohn**
Reverts istio/istio#45087

### CODE REVIEW

**Comment by rokkiter** [comment]
Do you think it is necessary to judge the slice length after split?

File: pilot/pkg/model/virtualservice.go [hunk 1]
```diff
@@ -601,10 +601,6 @@ func VirtualServiceDependencies(vs config.Config) []ConfigKey {
 			// shouldn't happen
 			continue
 		}
-		if len(knn) < 2 {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/virtualservice.go:604]: I think it is useful to prevent panic in L608

Comment (rokkiter) [reply pilot/pkg/model/virtualservice.go:604]: should i create a new pr after this merge?

**Comment by howardjohn** [comment]
I keep the VS part

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**